### ISSUE TITLE
ConfigConvert[Option[_]] doesn't work with None

### ISSUE THREAD

**Issue by melrief** [pureconfig/pureconfig#170]
I found a bug with the `Option` instance of `ConfigConvert`:

```scala
@ import pureconfig._ 
@ val cc = ConfigConvert[Option[Int]] 
@ cc.from(cc.to(None)) 
res3: Either[error.ConfigReaderFailures, Option[Int]] = Left(
  ConfigReaderFailures(
    CannotConvert("{}", "Int", """java.lang.NumberFormatException: For input string: "{}" """, None), List()))
```

The issue is that the `.from` method [expects `null` to read `None`](https://github.com/melrief/pureconfig/blob/master/core/src/main/scala/pureconfig/ConfigConvert.scala#L336) but `.to` [converts `None` into `{}`](https://github.com/melrief/pureconfig/blob/master/core/src/main/scala/pureconfig/ConfigConvert.scala#L344) instead of `null`. Fixing this is simple: just pass `null` to `ConfigValueFactory.fromAnyRef` in `ConfigConvert[Option].to` and everything work but I'm wondering if there is a reason why we are using `{}` for `None`.
Is this a bug or there is a good reason to not have `.from` and `.to` be the inverse of each other in the case of `None`?

**Comment by jcazevedo** [comment]
It's definitely a bug. We probably didn't notice it before because we rarely convert directly from `Option[_]` and we have special treatment for it in `hConsConfigConvert` so that we don't include the key in the resulting `ConfigObject`.

**Comment by melrief** [comment]
I will solve this inside #151 together with the tests that I've added for `Option[_]`

**Comment by melrief** [comment]
Solved with #183
